cirq-core/cirq/protocols/approximate_equality_protocol.py

```diff
@@ -141,6 +141,11 @@ def _approx_eq_iterables(val: Iterable, other: Iterable, *, atol: float) -> bool
         types.
     """
 
+    if isinstance(val, set):
+        val = sorted(val)
```

Comment:
This will throw TypeError for set `sa0 = {"a", 0}`, but such set should be approx_eq to itself - `approx_eq(sa0, sa0) is True`.


```suggestion
        try:
            val = sorted(val)
        except TypeError:
            return NotImplemented
```

and the same for `other` below.

You will also need to change the caller to correctly handle a `NotImplemented` value here -
https://github.com/quantumlib/Cirq/blob/078af27f2b7fd98fa64c2001e377fe8fac53eedb/cirq-core/cirq/protocols/approximate_equality_protocol.py#L116-L118